src/wp-includes/compat.php

```diff
@@ -375,6 +377,44 @@ function is_iterable( $var ) {
 	}
 }
 
+if ( ! function_exists( 'str_starts_with' ) ) {
+	/**
+	 * Polyfill for str_starts_with() function added in PHP 8.
+	 * Performs a case-sensitive check indicating if needle is contained in haystack.
+	 *
+	 * @ticket 54377
+	 * @ignore
+	 * @since 5.9.0
+	 *
+	 * @param $haystack string The string to search in.
+	 * @param $needle string The substring to search for in the haystack to is it startswith it.
+	 *
+	 * @return Boolean true if needle is the haystack and starts with it.
+	 */
+	function str_starts_with( $haystack, $needle ) {
+		return 0 === strpos( $haystack, $needle );
+	}
+}
+if ( ! function_exists( 'str_ends_with' ) ) {
+	/**
+	 * Polyfill for str_ends_with() function added in PHP 7.3.
+	 *
+	 * Performs a case-sensitive check indicating if needle is contained in haystack.
+	 *
+	 * @ticket 54377
+	 * @ignore
```

Comment:
- The `@ticket` annotation is not needed. 
- The `@ignore` annotation is not needed.